packages/server-wallet/src/wallet/store.ts

```diff
@@ -62,7 +65,19 @@ const throwMissingChannel: MissingAppHandler<any> = (channelId: string) => {
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 export class Store {
-  constructor(readonly timingMetrics: boolean, readonly skipEvmValidation: boolean) {
+  knex: Knex;
+
+  constructor(
+    readonly knexOrConfig: Knex | Knex.Config,
```

Comment:
I really think this should _insist_ on accepting a `Knex`.

We are now living in a world with a new "gotcha":
1. Say I create a `new Store` with a config object instead of a `Knex` instance.
2. Then, when I run my test, I will notice that jest doesn't exit.
3. I have to then learn that `Store` constructed a connection pool, and that pool may contain open connections, and that prevents the jest process from exiting.

There is a significant cost to (3).
A. Multiple people on this team have had to learn this the hard way, so this issue has already cost at least a full developer day.
B. When they do learn it, they need to maintain some additional overhead: "should I pass in a config? do I need to destroy the pool?"

I fail to see any benefit in the short-term of doing (1). Any benefit should be clearly stated, and (IMO) planned into our work towards consolidating the server and browser wallets.

Reply:
https://github.com/statechannels/statechannels/pull/2602